jax/experimental/array_serialization/serialization.py

```diff
@@ -184,6 +185,18 @@ async def release_bytes(self, requested_bytes):
       self._cv.notify_all()
 
 
+async def transfer_shard_to_host(shard: array.Shard) -> np.ndarray:
```

Comment:
Does this need to be async def? Why not just a normal function?

Reply:
Indeed, it's only the follow-up PR (https://github.com/google/jax/pull/22169) that we need this to be async.